src/http/parser.rs

```diff
@@ -0,0 +1,290 @@
+use std::collections::{HashMap, LinkedList};
```

Comment:
too much is going on in that file. Can we try split it to modules and add unit tests?